gateway/ln-gateway/src/lib.rs

```diff
@@ -90,7 +126,7 @@ pub struct Gateway {
 impl Gateway {
     #[allow(clippy::too_many_arguments)]
     pub async fn new(
-        lnrpc: DynLnRpcClient,
+        lnrpc: Arc<RwLock<dyn ILnRpcClient>>,
```

Comment:
I wonder if we should use https://crates.io/crates/arc-swap . I wish it was a part of stdlib, as it's a common use-case.

Reply:
Ack, but let me do this in the next refactor: https://github.com/fedimint/fedimint/issues/2069

I think once I get rid of the mut functions on `ILnRpcClient` it will become easier to use `ArcSwap`

Reply:
Actually, does `ArcSwap` only provide the ability to atomically swap out <T> (in this case, `ILnRpcClient`)? I think we actually need the ability to mutate `ILnRpcClient`. Currently we set the underlying client to None when the node is disconnected.

Reply:
`Arc<RwLock<dyn ILnRpcClient>>`) has no `Option`? Last time I checked `ILnRpcClient` had only `&self` methods, so other than ability to swap it whole, I don't see any mutability. Might have missed something.